packages/starknet-snap/src/state/network-state-manager.ts

```diff
@@ -88,10 +94,9 @@ export class NetworkStateManager extends StateManager<Network> {
   async updateNetwork(data: Network): Promise<void> {
```

Comment:
Is it intentional that a network can be retrieved from `availableNetworks` but not updated if it’s not in the state? If so, why is this behavior preferable?
If the goal is to ensure that any network retrieved can also be updated, should the state be updated with the network when it’s first accessed in `getNetwork` (if it’s not found)?
Otherwise we could comment this behavior, so future maintainers know that not all networks can be updated unless they exist in the state?

Reply:
or we can remove update method, as we dont have that use case

Reply:
I agree, let's remove this. If we need it we add it back. It then removes the confusion.

Reply:
just find out we have a use case 
we cant remove it actually

so update the comment on getNetwork instead